src/lib/FlightTasks/tasks/AutoLineSmoothVel/FlightTaskAutoLineSmoothVel.cpp

```diff
@@ -177,6 +177,50 @@ void FlightTaskAutoLineSmoothVel::_checkEkfResetCounters()
 	}
 }
 
+float FlightTaskAutoLineSmoothVel::_getSpeedAtTarget()
```

Comment:
can we move the two new methods into a UtilityClass/Function with more general names? then we can reuse it in other tasks as well

Reply:
Agreed, it would be great to use the `_getMaxSpeedFromDistance` in the collision prevention as well, instead of using just the velocity controller P gain.